docs/publishing.md

```diff
@@ -0,0 +1,139 @@
+# Publishing Guide
+
```

Comment:
we should separate dev docs vs user docs, not dump into same folder

it may make sense to have both in rtd, but separated there as well